bookkeeper-server/src/main/java/org/apache/bookkeeper/client/DistributionSchedule.java

```diff
@@ -149,6 +149,13 @@ public WriteSet copy() {
      */
     WriteSet getWriteSetForLongPoll(long entryId);
 
+    /**
+     * Return the set of bookies indices to send the messages to for force ledger.
+     *
+     * @return the set of bookies indices to send force request to.
+     */
+    WriteSet getWriteSetForForceLedger();
```

Comment:
This will become unmanageable. We ended up adding getWriteSetFor for different type of requests. I would suggest: renaming `getWriteSetForLongPoll` to `getEnsembleWriteSet`, because those requests are required to send to the whole ensemble.

Reply:
I like that idea; Everything that needs to get whole ensemble set can have one API. maybe call it `getEnsembleSet`, no need to have **write** in it.

Reply:
done.

I will use currentNonDurableLastAddConfirmed as reference entryId (prev getWriteSetForLongPoll required an entry id)
for DEFERRED_SYNC writers ensemble change is disabled, so it is not very important the entryId, but I think this is consistent with the fact that we want the ensemble at the current point in the history of the ledger